ga4gh/datamodel/__init__.py

```diff
@@ -488,3 +505,25 @@ def _getDataFilenames(self, dataDir, patterns):
 
     def getFileHandle(self, dataFile):
         return fileHandleCache.getFileHandle(dataFile, self.openFile)
+
+
+class MetadataSidecarMixin(object):
+    """
+    Loads a human readable sidecar and makes its values available as
+    self._sidecar(key). Accepts the filename from which we will load
+    the data.
+    """
+    def loadSidecar(self, filepath):
+        jsonFilename = os.path.splitext(filepath)[0] + ".json"
+        try:
+            with open(jsonFilename) as data:
+                self._sidecar = json.load(data)
+        except (ValueError, IOError):
+            raise exceptions.FileOpenFailedException(jsonFilename)
+            self._sidecar = {}
+
+    def sidecar(self, key):
```

Comment:
The problem is overloading the term metadata, in this case the actual data is represented by JSON so both metadata and sidecar are a bit obtuse. The goal here is to supply any datamodel object with a way to access key/values stored in a JSON. I'll refactor to follow the pattern you've requested.